Explorer/Assets/DCL/Navmap/NavmapSearchBarController.cs

```diff
@@ -0,0 +1,70 @@
+using Cysharp.Threading.Tasks;
+using DCL.AssetsProvision;
+using DCL.PlacesAPIService;
+using DCL.UI;
+using System;
+using System.Threading;
+
+namespace DCL.Navmap
+{
+    public class NavmapSearchBarController : IDisposable
+    {
+        public event Action<string> OnResultClicked;
+
+        private readonly SearchBarView view;
+        private readonly IPlacesAPIService placesAPIService;
+        private readonly SearchResultPanelController searchResultPanelController;
+
+        private CancellationTokenSource cts;

```

Comment:
Just like with any "proper" MVC Controller we should have a token for the whole disposal to prevent request from running on application exit